### PR TITLE
[Cache] fix expiration handling in ArrayAdapter

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

File: src/Symfony/Component/Cache/Adapter/ArrayAdapter.php [hunk 1]
```diff
@@ -137,6 +137,8 @@ public function save(CacheItemInterface $item)
         $expiry = $item[CacheItem::CAST_PREFIX.'expiry'];
 
         if (null !== $expiry && $expiry <= time()) {
+            $this->expiries[$key] = $expiry;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Cache/Adapter/ArrayAdapter.php]: `unset($this->values[$key], $this->expiries[$key]);`?

Comment (nicolas-grekas) [inline src/Symfony/Component/Cache/Adapter/ArrayAdapter.php]: or `return $this->deleteItem($key);`?

Comment (xabbuh) [inline src/Symfony/Component/Cache/Adapter/ArrayAdapter.php]: makes sense, fixed

**Comment by nicolas-grekas** [comment]
:+1:

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**